packages/openapi-ts/src/openApi/shared/transforms/readWrite.ts

```diff
@@ -581,10 +583,19 @@ export const updateRefsInSpec = ({
           });
         } else if (key === '$ref' && typeof value === 'string') {
           const map = split.mapping[value];
-          if (nextContext === 'read' && map?.read) {
-            (node as Record<string, unknown>)[key] = map.read;
-          } else if (nextContext === 'write' && map?.write) {
-            (node as Record<string, unknown>)[key] = map.write;
+          if (map) {
+            if (nextContext === 'read' && map.read) {
```

Comment:
Should be able to simplify this branching logic like this

```suggestion
            if (map?.read && (!nextContext || nextContext === 'read')) {
```